sdk/eventhub/Azure.Messaging.EventHubs.Processor/samples/Sample05_InitializeAPartition.cs

```diff
@@ -0,0 +1,162 @@
+                  Console.WriteLine($"Initialized partition: { eventArgs.PartitionId }");
+               }
+               catch (Exception ex)
+               {
+                   // For real-world scenarios, you should take action appropriate to your application.  For our example, we'll just log
+                   // the exception to the console.
+
+                   Console.WriteLine();
+                   Console.WriteLine($"An error was observed while initializing partition: { eventArgs.PartitionId }.  Message: { ex.Message }");
+                   Console.WriteLine();
+               }
+
+               return Task.CompletedTask;
+            }
+
+            // For this example, events will just be logged to the console.
+
+            Task processEventHandler(ProcessEventArgs eventArgs)
+            {
+               try
+               {
+                  Console.WriteLine($"Event Received: { Encoding.UTF8.GetString(eventArgs.Data.Body.ToArray()) }");
+               }
+               catch (Exception ex)
+               {
+                   // For real-world scenarios, you should take action appropriate to your application.  For our example, we'll just log
+                   // the exception to the console.
+
+                   Console.WriteLine();
+                   Console.WriteLine($"An error was observed while processing events.  Message: { ex.Message }");
+                   Console.WriteLine();
+               }
+
+               return Task.CompletedTask;
+            };
+
+            // For this example, exceptions will just be logged to the console.
+
+            Task processErrorHandler(ProcessErrorEventArgs eventArgs)
+            {
+                Console.WriteLine("===============================");
+                Console.WriteLine($"The error handler was invoked during { eventArgs.Operation ?? "Unknown" }, for Exception: { eventArgs.Exception.Message }");
+                Console.WriteLine("===============================");
+                Console.WriteLine();
+
+                return Task.CompletedTask;
+            }
+
+            processor.PartitionInitializingAsync += partitionInitializingHandler;
+            processor.ProcessEventAsync += processEventHandler;
+            processor.ProcessErrorAsync += processErrorHandler;
+
+            try
+            {
+                // In order to begin processing, an explicit call must be made to the processor.  This will instruct the processor to begin
+                // processing in the background, invoking your handlers when they are needed.
+
+                await processor.StartProcessingAsync();
+
+                // Because processing takes place in the background, we'll wait for a small period of time and then trigger
+                // cancellation.
+
+                using var cancellationSource = new CancellationTokenSource();
+                cancellationSource.CancelAfter(TimeSpan.FromSeconds(10));
```

Comment:
Partition claim happens in intervals of 10 seconds in the processor, so we'll probably end up initializing only one partition. Just telling you in case you think it's worth to wait 2 secs more and see if we can claim another one.

Reply:
Fair enough.   I think that's worth the investment.